lib/coach/handler.rb

```diff
@@ -76,11 +80,25 @@ def build_request_chain(sequence, context)
     end
 
     def inspect
-      "#<Coach::Handler[#{@root_item.middleware.name}]>"
+      "#<Coach::Handler[#{name}]>"
     end
 
     private
 
+    attr_reader :config
+
+    def root_item
+      @root_item ||= MiddlewareItem.new(middleware, config)
```

Comment:
I think we still want to validate the middleware?

```suggestion
      @root_item ||= MiddlewareItem.new(middleware, config).tap(&:validate!)
```

Reply:
Eek. Weird that no tests failed without this.

Reply:
Ok added some tests around this!